### PR TITLE
[SYCL][E2E] Update some XFAIL tests

### PR DESCRIPTION
**Pull request by KornevNikita**
Update the tests that already contain a link.
Also update lit config and tests to not include `.c` files, we don't have such tests in SYCL.

### CODE REVIEW

File: sycl/test/e2e_test_requirements/no-xfail-without-tracker.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@
 // - ...and check if the list of improperly XFAIL-ed tests needs to be updated.
 //
 // RUN: grep -rI "XFAIL:" %S/../../test-e2e \
-// RUN: -A 1 --include=*.c --include=*.cpp --no-group-separator | \
```
Comment (AlexeySachkov) [inline sycl/test/e2e_test_requirements/no-xfail-without-tracker.cpp:30]: I suggest that we also update `sycl/test-e2e/lit.cfg.py` to exclude `.c` suffix from being considered as a test

Comment (AlexeySachkov) [inline sycl/test/e2e_test_requirements/no-xfail-without-tracker.cpp:30]: And we need to make a similar change to our `unsupported` test, I assume

**Review by AlexeySachkov — APPROVED** [review]

**Review by Alcpz — APPROVED** [review]
SYCLcompat LGTM

**Review by aarongreig — APPROVED** [review]

File: sycl/test-e2e/ESIMD/assert.cpp [hunk 2]
```diff
@@ -8,6 +8,7 @@
 // as expected to fail, whilst it is being investigated, see intel/llvm#11359
 // FIXME: remove that XFAIL
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/assert.cpp:9]:
```suggestion
```
we can remove this line prolly too

**Review by sarnex — APPROVED** [review]

File: sycl/test-e2e/LLVMIntrinsicLowering/sub_byte_bitreverse.cpp [hunk 3]
```diff
@@ -4,8 +4,8 @@
 // UNSUPPORTED: hip || cuda
 
 // TODO: Remove XFAIL after fixing
```
Comment (aelovikov-intel) [inline sycl/test-e2e/LLVMIntrinsicLowering/sub_byte_bitreverse.cpp:6]: Likewise?

**Review by aelovikov-intel — APPROVED** [review]

**Comment by KornevNikita** [comment]
@intel/llvm-gatekeepers could you please merge?

**Pull request merged**